index.js

```diff
@@ -58,6 +63,28 @@ exports.getPostThread = function (ssb, mid, opts, cb) {
   })
 }
 
+exports.reviseFlatThread = function(ssb, thread, callback) {
+  // function that calls getLatestRevision on each of the elements of a flat
+  // thread, and collects them up into one callback.
+  var callbackAggregator = multicb({pluck: 1})
+
+  thread.forEach(function (thisMsg) { 
+    // we don't *need* to get the latest revision of an edit, it's included
+    // already
+    if (thisMsg.value.content.type === 'post') {
+      exports.getLatestRevision(ssb, thisMsg, callbackAggregator())
+    }
+  })
+  
+  callbackAggregator(function (err, results) {
+    if (err) {
+      callback(err)
+    } else {
+      callback(null, results)
+    }
+  })
+}
```

Comment:
oh wait, i see what you mean. there needs to be a passthru so other messages make it into the callback.